lib/inventory_manipulation.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# helper functions for manipulating and reading a bolt inventory file
+module InventoryManipulation
+  def inventory_hash_from_inventory_file(inventory_full_path = nil)
+    inventory_full_path = if inventory_full_path.nil?
+                            'inventory.yaml'
+                          else
+                            inventory_full_path
+                          end
+    raise "There is no inventory file at '#{inventory_full_path}'" unless File.exist?(inventory_full_path)
+
+    inventory_hash = YAML.load_file(inventory_full_path)
+    inventory_hash
+  end
+
+  def find_targets(inventory_hash, targets)
+    if targets.nil?
+      inventory = Bolt::Inventory.new(inventory_hash, nil)
+      targets = inventory.node_names.to_a
+    else
+      targets = [targets]
+    end
+    targets
+  end
+
+  def target_in_group(inventory_hash, node_name, group_name)
+    exists = false
+    inventory_hash['groups'].each do |group|
+      next unless group['name'] == group_name
+
+      group['nodes'].each do |node|
```

Comment:
Probably not the right time to do this, but:
```ruby
def target_in_group(inventory_hash, node_name, group_name)
  inventory_hash['groups'].any? do |group|
    group['name'] == group_name && group['nodes'].any? { node['name'] == node_name }
  end
end
```

Also, what's the whole pattern where it's a list with an entry name in it? `metadata.json` also has this madness. What's wrong with a hash? Then it's a simple `group = inventory_hash['groups'][group_name] ; !group.nil? && group.include?(node_name)`. Can probably also use `.try()` after it's dropped Puppet 4/Ruby 2.1.

Reply:
i'll have a look :) it may not go in this PR.

Reply:
Probably best not to do it in this PR :)